sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java

```diff
@@ -170,39 +171,33 @@ public void upload(InputStream data, long length, boolean overwrite) {
     public void uploadWithResponse(InputStream data, long length, ParallelTransferOptions parallelTransferOptions,
         BlobHttpHeaders headers, Map<String, String> metadata, AccessTier tier, BlobRequestConditions requestConditions,
         Duration timeout, Context context) {
-        uploadWithResponse(data, length, new BlobParallelUploadOptions()
+        uploadWithResponse(new BlobParallelUploadOptions(data, length)
             .setParallelTransferOptions(parallelTransferOptions).setHeaders(headers).setMetadata(metadata).setTier(tier)
-            .setRequestConditions(requestConditions), timeout, context);
+            .setRequestConditions(requestConditions).setTimeout(timeout), context);
     }
 
     /**
      * Creates a new blob, or updates the content of an existing blob.
      * <p>
      * To avoid overwriting, pass "*" to {@link BlobRequestConditions#setIfNoneMatch(String)}.
-     *
-     * @param data The data to write to the blob.
-     * @param length The exact length of the data. It is important that this value match precisely the length of the
-     * data provided in the {@link InputStream}.
      * @param options {@link BlobParallelUploadOptions}
-     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
      * @param context Additional context that is passed through the Http pipeline during the service call.
      * @return Information about the uploaded block blob.
      */
-    public Response<BlockBlobItem> uploadWithResponse(InputStream data, long length, BlobParallelUploadOptions options,
-        Duration timeout, Context context) {
-        BlobParallelUploadOptions blobParallelUploadOptions = options == null ? new BlobParallelUploadOptions()
-            : options;
+    public Response<BlockBlobItem> uploadWithResponse(BlobParallelUploadOptions options, Context context) {
```

Comment:
Does the corresponding async method have an options bag as well?

Reply:
Yes-ish. There is a corresponding upload method on async, but we did not add an options bag upload for it in master. We added it only on sync to address a Snowflake issue which required us to return the BlockBlobItem. I will be updating this for both sync and async in the stg73 branch. I can add the async overload here now, though, if you'd prefer it maintain its symmetry